doc/index.rst

```diff
@@ -55,6 +55,72 @@ you'll note that it runs much faster because it keeps track of virtualenv detail
 and will not recreate or re-install dependencies.  You also might want to
 checkout :doc:`examples` to get some more ideas.
 
+System overview
+---------------
+
+.. figure:: img/tox_flow.png
+   :align: center
+   :width: 800px
+
+   tox workflow diagram
+
+..
+    The above image raw can be found and edited by using the toxdevorg Google role account under
+    https://www.lucidchart.com/documents/edit/5d921f32-f2e1-4618-a265-7f9e30503dc6/0
+
+tox roughly follows the following phases:
+
+1. **configuration:** load ``tox.ini`` and merge it with options from the command line and the
+   operating system environment variables.
+2. **packaging** (optional): create a source distribution of the current project by invoking
+
+   .. code-block:: bash
+
+      python setup.py sdist
+
+   Note that for this operation the same Python environment will be used as the one tox is
+   installed into (therefore you need to make sure that it contains your build dependencies).
+   Skip this step for application projects that don't have a ``setup.py``.
+
+3. **environment** - for each tox environment (e.g. ``py27``, ``py36``) do:
+
+    1. **environment creation**: create a fresh environment, by default Python virtual environments
+    are used via the virtualenv_ project. tox will automatically
+    try to discover a valid Python interpreter version by using the environment name (e.g. ``py27``
+    means Python 2.7 and the ``basepython`` configuration value) and the current operating system
+    ``PATH`` value. This step is done at first run only (or when :confval:`basepython` value
+    changes) and re-used at subsequent runs, unless the user requests explicitly recreating this
+    by passing in the ``-r`` flag.
```

Comment:
A bit more handwavy but maybe sufficient here?

    This is created at first run only to be re-used at subsequent runs. If certain aspects of the project change, a re-creation of the environment is automatically triggered. To force the recreation (e.g. when changes weren't properly picked up), tox can be invoked with ``-r``/``--recreate``.

Reply:
I'll use most of it other than the `` (e.g. when changes weren't properly picked up)``. Some changes are not tracked, but I don't think that warrants it being stated not properly.